python/triton/compiler/compiler.py

```diff
@@ -48,6 +49,12 @@ def convert_type_repr(x):
         return '*' + convert_type_repr(match.group(1))
     return x
 
+class TritonPath(Path):
+    '''Adds MMAP R/O to Path "easy" read options'''
+    def mmap(self):
+        '''MMAP the file pointed to by path'''
+        return mmap.mmap(self.open(mode='rb').fileno, 0, prot=mmap.PROT_READ)
+
```

Comment:
Makes sense, but I don't see why we need a custom Path wrapper for this
```suggestion

def mmap_readonly(path: Path):
    return mmap.mmap(path.open(mode='rb').fileno, 0, prot=mmap.PROT_READ)

```